org-formation/090-systems-manager/Scheduled-Script-Automation.yaml

```diff
@@ -0,0 +1,308 @@
+            ExecutionRoleName: !Ref ExecutionRoleName
+            MasterAccountID: !Sub ${AWS::AccountId}
+            AutomationDocumentScriptExecution: !Ref AutomationDocumentScriptExecution
+            ScriptUrl: !Ref ScriptUrl
+            ScriptParameters: !Ref ScriptParameters
+            TargetTagName: !Ref TargetTagName
+            TargetTagValue: !Ref TargetTagValue
+            ExecutionTimeoutSeconds: !Ref ExecutionTimeoutSeconds
+      FunctionName: MultiAccountScriptExecution
+      Handler: index.handler
+      Role: !GetAtt AWSLambdaSSMMultiAccountScriptExecutionRole.Arn
+      Runtime: python3.12
+  AutomationAdministrationServiceRole:
+    Type: AWS::IAM::Role
+    Condition: CreateAutomationAdministrationRoleCondition
+    Properties:
+      RoleName: AWS-SystemsManager-AutomationAdministrationRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service: ssm.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: "/"
+      Policies:
+      - PolicyName: AssumeRole-AWSSystemsManagerAutomationExecutionRole
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - sts:AssumeRole
+            Resource:
+              Fn::Sub: arn:${AWS::Partition}:iam::*:role/AWS-SystemsManager-AutomationExecutionRole
+          - Effect: Allow
+            Action:
+            - organizations:ListAccountsForParent
+            Resource:
+            - "*"
+  AutomationDocumentScriptExecution:
+    Type: "AWS::SSM::Document"
+    Properties:
+      DocumentType: "Command"
+      Name: "Automation-Run-Script"
+      Content:
+        schemaVersion: "2.2"
+        description: "Run a parameterized script on EC2 instances"
+        parameters:
+          ScriptUrl:
+            type: String
+            description: "The URL of the script to run"
+          SsmParameterNameToEnvvarMap:
+            type: String
+            description: >-
+              space delimited list of environment name-value pairs in the format name:value,
+              e.g., ENVNAME1:envval1 ENVNAME2:envval2
+          TargetTagName:
+            type: String
+            description: name of the tag marking instances on which to run the script
+          TargetTagValue:
+            type: String
+            description: value of the tag marking instances on which to run the script
+          ExecutionTimeoutSeconds:
+            type: Number
+            description: Time limit for running the automation, in seconds
+            default: 120
+        mainSteps:
+        - name: "runShellScript"
+          action: "aws:runShellScript"
+          timeoutSeconds: {{ ExecutionTimeoutSeconds }}
+          onFailure: Abort
+          inputs:
+            Targets:
+            - Key: 'tag:{{ TargetTagName }}'
+              Values:
+                - '{{ TargetTagValue }}'
```

Comment:
for example, you are using the `Command` document `aws:runShellScript` here and from the [documentation](https://docs.aws.amazon.com/systems-manager/latest/userguide/documents-command-ssm-plugin-reference.html#aws-runShellScript) for it there is no `Targets:` parameter for `inputs`.    In fact the only place where I see `Targets` being available is for the [aws:runCommand action](https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-action-runcommand.html)

Reply:
As stated above, I think for this to work the Document has to be an `Automation` document.  I will update accordingly.